lib/oelite/cmd/bake.py

```diff
@@ -23,8 +28,46 @@ def parse_args(options, args):
     else:
         logging.getLogger().setLevel(logging.INFO)
 
+def os_check(config):
+    distro, release = setup.get_host_distro()
+    if distro and release:
+        setup_file, supported = setup.get_setup_file(config, distro, release)
+        if supported:
+            return True
+
+    if distro and release:
+        print('WARNING: Host distribution (%s %s) has not been validated'%(distro, release))
+    else:
+        print("WARNING: Cannot determine host distribution")
```

Comment:
this stuff does not really belong in bake. Let us make good use of OO and keep the responsibility for the logic about support host OS in the setup code.

What is needed is a refactor of the setup code, so you can simply ask it from bake "are we running on a supported host OS", probably something like a "is_supported()" method or the like. The rest of the setup code should of cause use that as well